python/Ganga/Core/GangaRepository/Registry.py

```diff
@@ -393,68 +345,50 @@ def checkShouldFlush(self):
             self._flushLastTime = timeNow
             self.dirty_hits = 0
 
-        self._lock.release()
-
         return decision
 
     def _getObjects(self):
         logger.debug("_getObjects")
-        #self._lock.acquire()
         returnable = self._objects
-        #self._lock.release()
         return returnable
 
     def ids(self):
```

Comment:
It probably should be locked but since it was already unlocked I was trying to not add any more locks than necessary. It will need a thorough review of the flow of information before we can go ahead and lock everything correctly.